### ISSUE TITLE
Floating point exception in tf.truncatemod when x is boundary value

### ISSUE THREAD

**Issue by DNXie** [tensorflow/tensorflow#45771]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):2.1.0
- Python version:3.7.6
- Bazel version (if compiling from source):N/A
- GCC/Compiler version (if compiling from source):N/A
- CUDA/cuDNN version:N/A
- GPU model and memory:N/A

**Describe the current behavior**
Floating point exception in `tf.truncatemod` when `x` is a boundary value of int64

**Describe the expected behavior**
expect no crash 

**Standalone code to reproduce the issue**
~~~python
import tensorflow as tf
tf.truncatemod(x=-9223372036854775808, y=[-1])
~~~

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with TF v2.3, TF v2.4 and TF-nightly. Please find the gist of it [here](https://colab.research.google.com/gist/amahendrakar/f05e9b4020b3aca1d08aa3c65f089f3b/45771-2-4.ipynb). Thanks!

**Comment by saikumarchalla** [comment]
@DNXie The Issue got fixed in TF-Nightly version(2.6). Closing the issue since it was resolved. Please find the gist [here](https://colab.research.google.com/gist/saikumarchalla/cf44b74f16ec85f49f6be35373f82bef/untitled90.ipynb).Thanks!
